server/bundler/utils.js

```diff
@@ -14,6 +14,9 @@ function getAssets( stats ) {
 }
 
 function pathToRegExp( path ) {
+	if ( path === '/' ) {
+		return path;
+	}
 	return new RegExp( '^' + path + '(/.*)?$' );
```

Comment:
> but that means changing all existing route definitions in wordpress-com and making sure they are kept in sync between the section definition and the route handling.

I feel like we should be able to generate the sections route list using the sections config, and perhaps a bit more mapping, or using additional convention in each section. 

I tested out changes here in the branch and verified that it fixes the double slash issue. I'm fine with 🚢ing this, but I would like us to take additional steps to address this in future PRs.